S/SIRIUS/build_tarballs.jl

```diff
@@ -0,0 +1,126 @@
+fi
+
+#need to pass various results of CMake's try_run() for MPI compilers for succesful build
+CMAKE_ARGS="${CMAKE_ARGS} -DMPI_RUN_RESULT_C_libver_mpi_normal=0 \
+                          -DMPI_RUN_RESULT_C_libver_mpi_normal__TRYRUN_OUTPUT='' \
+                          -DMPI_RUN_RESULT_CXX_libver_mpi_normal=0 \
+                          -DMPI_RUN_RESULT_CXX_libver_mpi_normal__TRYRUN_OUTPUT='' \
+                          -DMPI_RUN_RESULT_Fortran_libver_mpi_F90_MODULE=0 \
+                          -DMPI_RUN_RESULT_Fortran_libver_mpi_F90_MODULE__TRYRUN_OUTPUT='' \
+                          -DMPI_RUN_RESULT_Fortran_libver_mpi_F08_MODULE=0 \
+                          -DMPI_RUN_RESULT_Fortran_libver_mpi_F08_MODULE__TRYRUN_OUTPUT=''"
+
+cmake .. ${CMAKE_ARGS}
+
+#On MacOS, need to explicitly remove the -fallow-argument-mismatch flag, because not recognized by Clang
+if [[ "${target}" == *-apple* ]]; then
+   cmake .. "-DMPI_Fortran_COMPILE_OPTIONS=''"
+fi
+
+make -j${nproc} install
+"""
+
+augment_platform_block = """
+    using Base.BinaryPlatforms
+    $(MPI.augment)
+    augment_platform!(platform::Platform) = augment_mpi!(platform)
+"""
+platforms = [Platform("x86_64", "linux"), Platform("aarch64", "linux"), Platform("aarch64", "macos")]
+filter!(p -> !(libc(p) == "musl"), platforms)
+platforms = expand_cxxstring_abis(platforms)
+
+platforms = expand_gfortran_versions(platforms)
+filter!(p -> !(libgfortran_version(p) < v"5"), platforms)
+
+products = [
+   LibraryProduct("libsirius", :libsirius)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("GSL_jll"), 
+    Dependency("pugixml_jll"), 
+    #Using either MKL or OPENBLAS32
+    Dependency("libblastrampoline_jll"),
+    Dependency("Libxc_jll"), 
+    Dependency("HDF5_jll"),
+    Dependency("spglib_jll"), 
+    Dependency("spla_jll"), 
+    Dependency("SpFFT_jll"), 
+    Dependency("COSTA_jll"), 
+    Dependency("CompilerSupportLibraries_jll"), 
+    Dependency("LLVMOpenMP_jll", platforms=filter(Sys.isapple, platforms)),
+    HostBuildDependency(PackageSpec(; name="CMake_jll", version = v"3.28.1"))
+]
+
+platforms, platform_dependencies = MPI.augment_platforms(platforms; MPItrampoline_compat="5.2.1", OpenMPI_compat="4.1.6, 5")
+# Avoid platforms where the MPI implementation isn't supported
+# OpenMPI
+platforms = filter(p -> !(p["mpi"] == "openmpi" && arch(p) == "armv6l" && libc(p) == "glibc"), platforms)
+# MPItrampoline
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && (Sys.iswindows(p) || libc(p) == "musl")), platforms)
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && Sys.isfreebsd(p)), platforms)
+
+append!(dependencies, platform_dependencies)
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
+               augment_platform_block, julia_compat="1.6", preferred_gcc_version = v"10")
```

Comment:
libblastrampoline 5.4.0 necessarily requires at least julia v1.9:
```suggestion
               augment_platform_block, julia_compat="1.9", preferred_gcc_version = v"10")
```